### ISSUE TITLE
Support dynamic timezone values fed from a column in DATETRUNC

### ISSUE THREAD

**Issue by lakshmanan-v** [apache/pinot#8299]
In the current implementation DATETRUNC()'s optional timezone param is a hard coded string. We can pass only one value to it. In some use cases, each record in the table need to be supplied with its own timezone, coming from a different column (part of the fact table itself or coming from a dim table via lookup join).

It would be great if we can support this.

**Comment by xiangfu0** [comment]
Yeah, we should be able to modify `DateTruncTransformFunction` to take the timezone key from the dataSourceMap

**Comment by mayankshriv** [comment]
@KKcorps
